src/ProjectNeon/Assets/Features/Effects/QueueEffect.cs

```diff
@@ -0,0 +1,21 @@
+﻿
+/**
+ * Queues the decorated effect in BattleState effect queue.
+ * 
+ */
+public class QueueEffect : Effect
+{
+    private Effect _effect;
+
+    public QueueEffect(Effect effect)
+    {
+        _effect = effect;
+    }
+
+    public void Apply(Member source, Target target)
+    {
+        Effect effect = new UnqueueAfterExceuteEffect(_effect);
+        BattleEvent.Publish(new AddEffectToQueue(effect));
+        BattleEvent.Subscribe<EffectApplied>(_ => new RemoveEffectFromQueue(effect), this);
```

Comment:
This isn't checking which effect was applied. We need to use the `EffectApplied` message and ensure that this the correct effect.

Also, we are missing the corresponding Unsubscribe. It can happen inside the delegate right here.